web/packages/teleport/src/Discover/Shared/usePoll.ts

```diff
@@ -24,16 +21,6 @@ export function usePoll<T>(
     }
   }, [callback, enabled, running]);
```

Comment:
Why is `callback` needed in this sensitivity list?

Reply:
In fact, can we not simplify this whole hook to be

```typescript
export function usePoll<T>(
  callback: (signal: AbortSignal) => Promise<T | null>,
  enabled: boolean,
  interval = 1000
): T | null {
  const abortController = useRef(new AbortController());
  const [result, setResult] = useState<T | null>(null);

  useEffect(() => {
    if (enabled) {
      abortController.current = new AbortController();

      const id = window.setInterval(async () => {
        try {
          const result = await callback(abortController.current.signal);

          if (result) {
            clearInterval(id);
            setResult(result);
          }
          // eslint-disable-next-line no-empty
        } catch {}
      }, interval);

      return () => {
        clearInterval(id);
        abortController.current.abort();
      };
    } else {
      setResult(null);
    }
  }, [enabled, interval, callback]);

  return result;
}
```

Reply:
actually nvm, i think i can make use of the `enabled` flag

Reply:
Yeh I think so, I've updated it to use yours